hooks/cmd/install/install.go

```diff
@@ -0,0 +1,225 @@
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package main
+
+import 	(
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+        "log/syslog"
+	"os"
+	"strconv"
+
+	"github.com/canonical/app-service-configurable/hooks"
+)
+
+var log syslog.Writer
+
+func handleDirs() error {
+	dataConfP := fmt.Sprintf("%s/config/res", hooks.SnapDataE)
+	snapConfP := fmt.Sprintf("%s/config/res", hooks.SnapE)
+
+	// install all the config files from $SNAP/config/res/ into $SNAP_DATA/config/res/,
+	// but if files already exist, don't over-write
+	log.Info(fmt.Sprintf("edgex-asc:install creating: %s", dataConfP))
+	err := os.MkdirAll(dataConfP, 0755)
+	if err != nil {
+		return errors.New(fmt.Sprintf("Can't make %s - %v", dataConfP, err))
+	}
+
+	// create profile dirs if needed
+	// open snapConfP dir
+	// call os.Readdir, returns a slice of FileInfo
+	// walk slice, creating directories for all IsDir==true
+	f, err := os.Open(snapConfP)
+	if err != nil {
+		return errors.New(fmt.Sprintf("Can't open %s - %v", snapConfP, err))
+	}
+	defer f.Close()
+
+	files, err := f.Readdir(0)
+	if err != nil {
+		return errors.New(fmt.Sprintf("Can't read %s dir - %v", snapConfP, err))
+	}
+
+	for _, fi := range files {
+		log.Info(fmt.Sprintf("edgex-asx:install: walking files in $SNAP/config/res - %s", fi.Name()))
+		if fi.IsDir() {
+			name := fi.Name()
+
+			// TODO: review sample configuration dups w/upstream
+			if name == "sample" {
+				continue
+			}
+
+			destD := fmt.Sprintf("%s/%s", dataConfP, name)
+			err = os.Mkdir(destD, 0755)
+			if err != nil {
+				return errors.New(fmt.Sprintf("Can't create dir %s - %v", destD, err))
+			}
+
+			srcP := fmt.Sprintf("%s/%s/configuration.toml", snapConfP, name)
+			sf, err := os.Open(srcP)
+			if err != nil {
+				return errors.New(fmt.Sprintf("Can't open %s - %v", srcP, err))
+			}
+			defer sf.Close()
+
+			destP := fmt.Sprintf("%s/configuration.toml", destD)
+			log.Info(fmt.Sprintf("edgex-asc:install creating: %s", destP))
+			df, err := os.OpenFile(destP, os.O_WRONLY|os.O_CREATE, 0755)
+			if err != nil {
+				return errors.New(fmt.Sprintf("Can't create %s - %v", destP, err))
+			}
+			defer df.Close()
+
+			_, err = io.Copy(df, sf)
+			if err != nil {
+				return errors.New(fmt.Sprintf("Can't copy %s to %s - %v", srcP, destP, err))
+			}
+		}
+	}
+
+	return nil
+}
+
+// TODO: merge w/configure version
+func handleVal(p string, k string, v interface{}, flatConf map[string]interface{}) {
+	var mk string
+
+	// top level keys don't include "env", so no separator needed
+	if p == "" {
+		mk = k
+	} else {
+		mk = fmt.Sprintf("%s.%s", p, k)
+	}
+
+	log.Info(fmt.Sprintf("handleVal: mk: %s", mk))
+
+	switch t := v.(type) {
+	case string:
+		log.Info(fmt.Sprintf("ADDING %s=%s to flatConf", k, t))
+		flatConf[mk] = t
+	case bool:
+		log.Info(fmt.Sprintf("ADDING %s=%v to flatConf", k, t))
+		flatConf[mk] = strconv.FormatBool(t)
```

Comment:
why not just 

```suggestion
		flatConf[mk] = fmt.Sprintf("%t", t)
```
and similar for below?

Reply:
Because I'd read that strconv was much more performant than Sprintf, especially in the case where you're converting a single value vs. a value embedded in a longer string...

Reply:
Ah fair enough it probably is, feel free to leave it as is in that case then